prow/cmd/gerrit/main.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// package main is a gerrit fetcher updating against a target gerrit instance
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"os/signal"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/gerrit"
+	"k8s.io/test-infra/prow/kube"
+)
+
+var (
+	configPath = flag.String("config-path", "/etc/config/config", "Path to config.yaml.")
+
+	instance  = flag.String("gerrit-instance", "", "URL to gerrit instance")
+	projects  = flag.String("gerrit-projects", "", "comma separated gerrit projects to fetch from the gerrit instance")
+	rateLimit = flag.Int("rate-limit", 5, "How many gerrit changes to fetch per API call")
+)
+
+func main() {
+	flag.Parse()
+	logrus.SetFormatter(&logrus.JSONFormatter{})
+	logger := logrus.WithField("component", "gerrit")
+
+	if *instance == "" {
+		logger.Fatal("--gerrit-intance must set")
+	}
+
+	projs := strings.Split(*projects, ",")
+	if len(projs) == 0 {
+		logger.Fatal("must have one or more target gerrit project")
+	}
+
+	// authenticate to gerrit server
+	if err := exec.Command("git", "clone", "https://gerrit.googlesource.com/gcompute-tools").Run(); err != nil {
```

Comment:
hmm, seems like we can do this without a git clone at runtime, why not require this tool to be present and include it in the image? this smells a bit.

Reply:
yeah, I guess I'm not sure how to do this with the prow bazel rules, or maybe I can toss up a Dockerfile with a runner?

Reply:
with bazel you could:
- seperately publish a base image with this and reference it as your base layer
- make this tool a WORKSPACE dep or check it in and then just copy it into the bazel built image

Reply:
it looks like these are two infrequently modified python scripts, I would just vendor the repo and copy it into the image when building

Reply:
ack, so I need to manually `dep add` and `dep ensure` or something? (lost on dep again, probably talk to you tomorrow)

Reply:
not sure if dep can handle go code? but if not we can put it in a third_party dir or something.

Reply:
yeah, `dep` and `vendor/` are only for Go.

a `third_party` directory would work, but adding a new `git_repository` to the bazel `WORKSPACE` might be even better.

Reply:
the downside to a `git_repository` is needing another checkout between every clean, I think we probably don't want/need that for ~300 lines of python and a license/readme